File: cpp/src/valid_anagram.cpp
#include <string>
#include <unordered_map>

using namespace std;

class Solution {
public:
    bool isAnagram(string s, string t) {
        unordered_map<char, int> letters;
 
        if (s.length() == t.length()){
            for (char l : s){
                if (letters.count(l)) {
                    letters[l]++;
                    continue;
                }
                letters[l] = 1;
            }

            for (char l : t){
                if (!letters.count(l)){
                    return false;
                }

                letters[l] -= 1;
                if (letters[l] < 0) {
                    return false;
                }
            }
            
            return true;
        }

        return false;
    }
};

File: python/valid_anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) == len(t):
            letters = {}

            for l in s:
                if l in letters:
                    letters[l] += 1
                else:
                    letters[l] = 1

            for l in t:
                if l not in letters:
                    return False

                letters[l] -= 1

            for v in letters.values():
                if v:
                    return False

            return True

        return False